### PR TITLE
Fail contacts on Vumi and Twitter who have opted out

### PR DESCRIPTION
**Pull request by nicpottier**
Automatically fails contacts on Vumi and Twitter who we know either no longer exist or who have explicitly opted out by unfollowing or sending "stop"

### CODE REVIEW

**Comment by nicpottier** [comment]
plz review @rowanseymour

File: temba/channels/models.py [hunk 1]
```diff
@@ -1473,7 +1479,26 @@ def send_twitter_message(cls, channel, msg, text):
...
-        dm = twitter.send_direct_message(screen_name=msg.urn_path, text=text)
+        try:
+            dm = twitter.send_direct_message(screen_name=msg.urn_path, text=text)
+        except Exception as e:
+            error_code = getattr(e, 'error_code', 400)
+            fatal = False
+
+            # this handle doesn't exist anymore or we can't send to them, fail them
+            if error_code == 404 or \
+              (error_code == 403 and str(e).index('users who are not following you')):
```
Comment (rowanseymour) [inline temba/channels/models.py]: Searching for that string is obviously a little fragile here but assuming there's no other way to be sure what the 403 is about

**Comment by rowanseymour** [comment]
Yarp

File: temba/channels/models.py (hunk 1 above)
Comment (nicpottier) [inline temba/channels/models.py]: Ya, there aren't any more details passed up from Twython.

**Pull request merged**